src/pynxtools/dataconverter/convert.py

```diff
@@ -398,9 +406,9 @@ def convert_cli(
             nxdl,
             output,
             skip_verify,
-            config_file=config,
             ignore_undocumented=ignore_undocumented,
             fail=fail,
+            **kwargs,
```

Comment:
This doesn't work because `config_file` is not in kwargs (it is an input to this function). We can simply add it to kwargs if it exists:
```
if config_file:
    kwargs["config_file"] = config_file
```
Actually, this whole discussion is part of a bigger problem, namely that we call 
```
data = data_reader().read(  # type: ignore[operator]
        template=Template(template), file_paths=input_file, **kwargs
    )
```
in the `convert` function. But, most of our reader don't actually accept kwargs, so this will always be a bit unstable.

Reply:
I added 
```
if config_file:
    kwargs["config_file"] = config_file
```
to the `convert_cli` function.

Reply:
Good find. That fix works. Thank you.

If I remember correctly, Python expands all values of kwargs automatically. So it will pass what it receives. So the signature of the read function in the reader should match the parameters we sent to the cli during invocation. If we run into more errors, we can have this more thoroughly controlled before it's passed to the readers.

Reply:
Yeah, for now it works. I guess at some point we could declutter these things, but for it's fine.

Reply:
Something like `func_name.__code__.co_varnames` should give you a list of possible keyword arguments, which you can use for raising a suitable error message.